### ISSUE TITLE
DeprecationWarning using segmentation.slic

### ISSUE THREAD

**Issue by ntjess** [scikit-image/scikit-image#4822]
## Description
The example code for `slic` results in a DeprecationWarning.

## Way to reproduce
```python
from skimage.segmentation import slic
from skimage.data import astronaut
img = astronaut()
segments = slic(img, n_segments=100, compactness=10)
```
Output:
```
Traceback (most recent call last):
  File "<input>", line 4, in <module>
  File "C:\Users\ntjes\miniconda3\envs\py38\lib\site-packages\skimage\segmentation\slic_superpixels.py", line 206, in slic
    image = img_as_float(image)
  File "C:\Users\ntjes\miniconda3\envs\py38\lib\site-packages\skimage\util\dtype.py", line 452, in img_as_float
    return _convert(image, np.floating, force_copy)
  File "C:\Users\ntjes\miniconda3\envs\py38\lib\site-packages\skimage\util\dtype.py", line 226, in _convert
    dtypeobj_out = np.dtype(dtype)
DeprecationWarning: Converting `np.inexact` or `np.floating` to a dtype is deprecated. The current result is `float64` which is not strictly correct.

```

## Version information
```python
# Paste the output of the following python commands
from __future__ import print_function
import sys; print(sys.version)
import platform; print(platform.platform())
import skimage; print("scikit-image version: {}".format(skimage.__version__))
import numpy; print("numpy version: {}".format(numpy.__version__))
```

```python
3.8.3 (default, Jul  2 2020, 17:30:36) [MSC v.1916 64 bit (AMD64)]
Windows-10-10.0.18362-SP0
scikit-image version: 0.17.2
numpy version: 1.19.0
```

**Comment by emmanuelle** [comment]
Hi @ntjess thanks for reporting this issue. Unfortunately I can't reproduce on my Linux machine. Could you please try to see if 
```python
from skimage import img_as_float
from skimage.data import astronaut
img = astronaut()
img = img_as_float(img)
```
results in the same warning? I'm trying to narrow down the problem.

**Comment by ntjess** [comment]
I see now that this came up during my unit tests because I installed a warning filter:

```python
from warnings import simplefilter
simplefilter('always')
```

The code you pasted should show an error at that point.

**Comment by cameronbronstein** [comment]
I'm also getting a similar warning, routed through use of [skimage/util/dtype.py:226](https://github.com/scikit-image/scikit-image/blob/3a4dc584d062347d9ab6bcd0910aeb7b40bd9623/skimage/util/dtype.py#L226) by Rasterio: ``DeprecationWarning: Converting `np.inexact` or `np.floating` to a dtype is deprecated. The current result is `float64` which is not strictly correct.
    dtypeobj_out = np.dtype(dtype)```

Anyway, this warning comes when running unit-tests. Nothing explicitly using/testing anything from skimage.

**Comment by bilderbuchi** [comment]
Related/dupe of #4848? does #4731 fix the issue?

**Comment by sciunto** [comment]
Seems to be fixed in master (and will be backported in 0.17.3 if released). I'm closing. If you still encounter the issue in master or in 0.17.3, please reopen.
